include/aws/common/system_info.h

```diff
@@ -100,6 +100,29 @@ void aws_backtrace_print(FILE *fp, void *call_site_data);
 AWS_COMMON_API
 void aws_backtrace_log(int log_level);
 
+/**
+ * Sets the passed double value as the current CPU usage. Only supported on Linux currently.
+ */
+AWS_COMMON_API
+int aws_get_cpu_usage(
+    uint64_t *cpu_last_total_user,
+    uint64_t *cpu_last_total_user_low,
+    uint64_t *cpu_last_total_system,
+    uint64_t *cpu_last_total_idle,
+    double *output);
+
+/**
+ * Sets the passed double value as the memory usage in kilobytes. Only supported on Linux currently.
+ */
+AWS_COMMON_API
+int aws_get_memory_usage(double *output);
+
+/**
+ * Sets the passed double value as the process count. Only supported on Linux currently.
+ */
+AWS_COMMON_API
+int aws_get_process_count(double *output);
```

Comment:
Can we make both of these uint64s?  If we have to convert to double later because of JSON, that's fine, but a general query in aws-c-common shouldn't be fractional